vcfanno.go

```diff
@@ -128,29 +128,32 @@ see: https://github.com/brentp/vcfanno
 
 	var err error
 	var qrdr io.Reader
-	// try to parallelizing reading if we have plenty of CPUs and it's (possibly)
+	// try to parallelize reading if we have plenty of CPUs and it's (possibly)
 	// a bgzf file.
-	if len(config.Annotation) < runtime.GOMAXPROCS(0) && strings.HasSuffix(queryFile, ".gz") {
+	if len(config.Annotation) < runtime.GOMAXPROCS(0) && strings.HasSuffix(queryFile, ".gz") || strings.HasSuffix(queryFile, ".bgz") {
 		if rdr, err := os.Open(queryFile); err == nil {
 			if st, err := rdr.Stat(); err == nil && st.Size() > 2320303098 {
 				qrdr, err = bgzf.NewReader(rdr, 4)
 				if err == nil {
-					log.Printf("using 4 worker threads to decompress query file")
+					log.Printf("using 4 worker threads to decompress bgzip file")
+				} else {
+					qrdr = nil
 				}
-				qrdr = nil
```

Comment:
Without an else clause here, when bgzip reader succeeds OR fails, qrdr is unconditionally being set to nil, thus always falls back to non-bgzip reader (xopen)